src/test/java/hudson/plugins/ec2/win/winrm/WinRMClientWithFIPSTest.java

```diff
@@ -0,0 +1,91 @@
+package hudson.plugins.ec2.win.winrm;
+
+import static org.junit.Assert.fail;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import jenkins.security.FIPS140;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.FlagRule;
+
+public class WinRMClientWithFIPSTest {
+
+    @ClassRule
+    public static FlagRule<String> fipsSystemPropertyRule =
+            FlagRule.systemProperty(FIPS140.class.getName() + ".COMPLIANCE", "true");
+
+    /**
+     * Self-signed certificate should not be allowed in FIPS mode, an {@link IllegalArgumentException} is expected
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void testSelfSignedCertificateNotAllowed() throws MalformedURLException {
+        new WinRMClient(new URL("https://localhost"), "username", "password", true);
+    }
```

Comment:
NIT, the modern way of testing an expected exception is `assertThrows(Throwable.class, () -> ...)`

Reply:
Done: 22bda8fb5d07f38e51cb0e403dc6afb7a87e250d